toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -70,52 +78,47 @@ func main() {
 
-	err = fetchPackages()
+	dependencyGraph, err := pkggraph.ReadDOTGraphFile(*inputGraph)
 	if err != nil {
-		logger.Log.Fatalf("Failed to fetch packages. Error: %s", err)
+		logger.Log.Fatalf("Failed to read graph to file: %s", err)
+		return
 	}
-}
 
-func fetchPackages() (err error) {
-	var cloner *rpmrepocloner.RpmRepoCloner
-	dependencyGraph, err := pkggraph.ReadDOTGraphFile(*inputGraph)
+	hasUnresolvedNodes := hasUnresolvedNodes(dependencyGraph)
+	if hasUnresolvedNodes || *tryDownloadDeltaRPMs {
+		err = fetchPackages(dependencyGraph, hasUnresolvedNodes, *tryDownloadDeltaRPMs)
+		if err != nil {
+			logger.Log.Fatalf("Failed to fetch packages. Error: %s", err)
+		}
+	}
+
+	// Write the final graph to file
+	err = pkggraph.WriteDOTGraphFile(dependencyGraph, *outputGraph)
 	if err != nil {
-		err = fmt.Errorf("failed to read graph to file:\n%w", err)
+		logger.Log.Fatalf("Failed to write cache graph to file: %s", err)
 		return
 	}
+}
 
-	toolchainPackages, err := schedulerutils.ReadReservedFilesList(*toolchainManifest)
+func fetchPackages(dependencyGraph *pkggraph.PkgGraph, hasUnresolvedNodes, tryDownloadDeltaRPMs bool) (err error) {
+	// Create the worker environment
+	cloner, err := setupCloner()
 	if err != nil {
-		err = fmt.Errorf("unable to read toolchain manifest file '%s':\n%w", *toolchainManifest, err)
+		err = fmt.Errorf("failed to setup cloner:\n%w", err)
 		return
 	}
+	defer cloner.Close()
 
-	hasUnresolvedNodes := hasUnresolvedNodes(dependencyGraph)
 	if hasUnresolvedNodes {
-		// Create the worker environment
-		cloner, err = rpmrepocloner.ConstructCloner(*outDir, *tmpDir, *workertar, *existingRpmDir, *existingToolchainRpmDir, *tlsClientCert, *tlsClientKey, *repoFiles)
+		var toolchainPackages []string
+		logger.Log.Info("Found unresolved packages to cache, downloading packages")
+		toolchainPackages, err = schedulerutils.ReadReservedFilesList(*toolchainManifest)
 		if err != nil {
-			err = fmt.Errorf("failed to setup new cloner:\n%w", err)
-			return err
-		}
-		defer cloner.Close()
-
-		enabledRepos := rpmrepocloner.RepoFlagAll
-		if !*usePreviewRepo {
-			enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagPreview
-		}
-		if *disableUpstreamRepos {
-			enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagUpstream
-		}
-		if *disableDefaultRepos {
-			enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagMarinerDefaults
+			err = fmt.Errorf("unable to read toolchain manifest file '%s':\n%w", *toolchainManifest, err)
+			return
 		}
-		cloner.SetEnabledRepos(enabledRepos)
-	}
 
-	if hasUnresolvedNodes {
-		logger.Log.Info("Found unresolved packages to cache, downloading packages")
-		err = resolveGraphNodes(dependencyGraph, *inputSummaryFile, *outputSummaryFile, toolchainPackages, cloner, *stopOnFailure)
+		err = resolveGraphNodes(dependencyGraph, *inputSummaryFile, toolchainPackages, cloner, *stopOnFailure)
 		if err != nil {
 			err = fmt.Errorf("failed to resolve graph:\n%w", err)
 			return err
```

Comment:
>return err [](http://example.com/codeflow?start=3&length=10)

Nit-pick: let's consistently return with `return` or `return err`. In general, I'd stick to using `return` and leave `return err` for special cases, where we re-define `err` within a smaller scope with `err := ...`.